src/cbapi/live_response_api.py

```diff
@@ -985,9 +985,21 @@ def _spawn_new_workers(self):
 
         schedule_max = self._max_workers - len(self._job_workers)
 
-        sensors = [s for s in self._cb.select(Sensor) if s.id in self._unscheduled_jobs
-                   and s.id not in self._job_workers
-                   and s.status == "Online"]
+        sensors = []
+
+        for sensor_id in self._unscheduled_jobs:
+            sensor = self._cb.select(Sensor, sensor_id)
+            if sensor_id in self._job_workers:
+                log.debug("Skipping sensor {} already has a worker created".format(sensor_id))
+            elif sensor.status == "Online":
+                sensors.append()
```

Comment:
shouldn't it be sensors.append(sensor) ? sensors.append() does not append anything to the sensors list

Reply:
sorry, i'm wrong, this is fixed in the second commit

Reply:
Thanks! I had forgot that and it crashed on me during my test :)